src/main/scala/com/typesafe/sbt/site/SitePreviewPlugin.scala

```diff
@@ -34,11 +41,23 @@ object SitePreviewPlugin extends AutoPlugin {
       server stop()
       server destroy()
     },
+    previewAuto := {
+      val port = previewFixedPort.value getOrElse Port.any
+      val sLog = streams.value.log
+      val browser = previewLaunchBrowser.value
+
+      Preview(port, (target in previewAuto).value, makeSite, watchSources, state.value) run { server =>
+        if(browser)
+          Browser open(server.portBindings.head.url)
+      }
+      sLog.info("SitePreviewPlugin server started on port %d. Press any key to exit." format port)
```

Comment:
yeah it's unfiltered: https://github.com/unfiltered/unfiltered/blob/2e10ca63fb4cd49daef972fab10642ad7543b4f5/util/src/main/scala/servers.scala#L75-L78